CMakeLists.txt

```diff
@@ -1,9 +1,18 @@
 if(USE_GPU OR APPLE)
   cmake_minimum_required(VERSION 3.2)
+elseif(USE_CUDA)
+  cmake_minimum_required(VERSION 3.11)
+  enable_language(CUDA)
 else()
   cmake_minimum_required(VERSION 2.8)
 endif()
 
+if(USE_CUDA)
+  PROJECT(lightgbm LANGUAGES C CXX CUDA)
+else()
+  PROJECT(lightgbm LANGUAGES C CXX)
+endif()
+
 PROJECT(lightgbm)
```

Comment:
Actually I was talking about line 16 (unconditioned `PROJECT(lightgbm)` command). Could you please add back 
```
if(USE_CUDA)
  PROJECT(lightgbm LANGUAGES C CXX CUDA)
else()
  PROJECT(lightgbm LANGUAGES C CXX)
endif()
```
construction? I find it very clear and it will eliminate multiple `PROJECT` definitions for CUDA case.